core/src/main/java/org/apache/calcite/sql/SqlCreateOrReplace.java

```diff
@@ -0,0 +1,45 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to you under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ *
+ */
+package org.apache.calcite.sql;
+
+/**
+ * Enum used by {@link org.apache.calcite.sql.SqlCreate} to indicate whether the query is a
+ * "CREATE", "CREATE OR REPLACE", or "REPLACE" query
+ */
+public enum SqlCreateOrReplace {
```

Comment:
I think "CreateOrReplace" is too ambiguous of a name for this enum. It can be misinterpreted as having either of the two meanings:
- "CREATE OR REPLACE"
- "CREATE" OR "REPLACE"

As such, I think we should steer away from that ambiguity and select something slightly more explicit. How about "enum SqlCreateType" or "enum SqlCreateOption"?